opengrok-indexer/src/main/java/org/opengrok/indexer/web/Laundromat.java

```diff
@@ -78,25 +79,24 @@ public static String launderLog(String value) {
      * collisions, the colliding keys' values are combined.
      */
     public static Map<String, String[]> launderLog(Map<String, String[]> map) {
-        if (map == null) {
-            return null;
-        }
-
         HashMap<String, String[]> safes = new HashMap<>();
-        for (Map.Entry<String, String[]> entry : map.entrySet().stream().sorted(
-                Map.Entry.comparingByKey()).collect(Collectors.toList())) {
-            String key = launderLog(entry.getKey());
-            String[] safeValues = safes.get(key);
-            String[] fullySafe = mergeLogArrays(entry.getValue(), safeValues);
-            safes.put(key, fullySafe);
-        }
+
+        Optional.ofNullable(map)
+                .stream()
+                .map(Map::entrySet)
+                .flatMap(Collection::stream)
+                .sorted(Map.Entry.comparingByKey())
+                .forEach(entry ->
+                        safes.compute(
+                                launderLog(entry.getKey()),
+                                (k, v) -> mergeLogArrays(entry.getValue(), v)
```

Comment:
Could `v` be `null` ? Would it be better to use `safes.merge()` ?

Reply:
Updated to use merge approach using Collectors.toMap